RocketChat/Rocket.Chat
# fix: users without field roles in db

## Proposed changes (including videos or screenshots)

Fix the problem of users not having the field `roles` in database when the default role setting is not set.




## Issue(s)


## Steps to test or reproduce
1. Go to workspace > configuration > accounts > register;
 Set the Registration Form as `Secret URL`;
 Set the Default Roles for Users field as blank;
2. Using the registration url, create a user (`https://[workspace_url]/register/[secret_hash]`)
3. With an admin user, go to [workspace > users] and try to check the created user info;

A grey screen with an error will be shown since the user doesn't have the `roles` field. It also possible to check via DB in the `users` collection.


## Further comments

[SUP-207](https://rocketchat.atlassian.net/browse/SUP-207)


[SUP-207]: https://rocketchat.atlassian.net/browse/SUP-207?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
